interface/backend/submission.py

```diff
@@ -11,10 +16,24 @@
 def handle_submission(file, assignment, user):
     log.debug(f'Submission {file.name} received')
 
-    submission = assignment.submission_set.create(
-        user=user,
-        archive_size=file.size,
-    )
+    with transaction.atomic():
+        Assignment.objects.select_for_update().get(pk=assignment.pk)
+        entries = Submission.objects.filter(user=user).order_by('-timestamp')
```

Comment:
But since we're locking the assignment, shouldn't we be checking only uploads for this assignment?

`assignment.submission_set.filter(user=user)...`